### PR TITLE
Make GN and IN the same dtype behavior as BN or LN in mixed_precision

### LINKED ISSUE #2550: GroupNorm + FP16 instability
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Colab
- TensorFlow version and how it was installed (source or binary): 2.6.0
- TensorFlow-Addons version and how it was installed (source or binary): 0.14.0
- Python version: 3
- Is GPU used? (yes/no): no

**Describe the bug**
In some cases GroupNormalization layer generates Inf/Nan values with mixed_fp16 policy.

**Code to reproduce the issue**
https://colab.research.google.com/drive/17zqRy30dbOcJpWZfOwZK3UFovIJZCuqb?usp=sharing

### PR DESCRIPTION
**Pull request by Zhaopudark**
This PR makes GroupNorm and InstanceNorm the same dtype behavior as keras' BatchNorm or LayerNorm in mixed_precision.

Fixes #2550

## Type of change

- [x] Bug fix

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [x] By running Black + Flake8
    - [x] By running pre-commit hooks
- [x] This PR addresses an already submitted issue for TensorFlow Addons
- [x] I have added tests that prove my fix is effective or that my feature works

# How Has This Been Tested?
` python -m pytest .\tensorflow_addons\layers\tests\normalizations_test.py`
Add accuracy tests for modified GN by setting tf.keras.layers.LayerNormalization as baseline.

### CODE REVIEW

**Comment by bhack** [comment]
@shkarupa-alex Are you still interested in this?

**Comment by shkarupa-alex** [comment]
> @shkarupa-alex Are you still interested in this?

Yes, i'm interested.
But this PR https://github.com/tensorflow/tensorflow/pull/52217 fixes my previous issue https://github.com/tensorflow/addons/issues/2550

Just waiting to merge.

But it would be nice to have a workaround while waiting for that.

**Comment by seanpmorgan** [comment]
Thank you for your contribution. We sincerely apologize for any delay in reviewing, but TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
